recipes/util-linux/util-linux-2.29/hwclock-tweak-delay.patch

```diff
@@ -0,0 +1,100 @@
+hwclock: allow tweaking hard-coded .5s in --systohc algorithm
+
+The algorithm hwclock uses for setting the RTC from the system clock
+uses "knowledge" about the hardware, which unfortunately only applies
+to some hardware (x86-like platforms, in particular). On other
+platforms where the fractional part of the RTC is cleared when the RTC
+is set, this algorithm causes a consistent .5 second loss when doing
+hwclock --systohc.
+
+So allow overriding those hard-coded .5. To figure out how a specific
+platform behaves, and thus whether one should use '--delay 0.0', one
+can do something like
+
+# Make sure ntpd/chronyd/ are down.
+# Set system clock once.
+ntpdate 0.pool.ntp.org 1.pool.ntp.org
+# Write the system time to the RTC, read it back, compare to NTP time.
+hwclock --utc --systohc --noadjfile
+hwclock --utc --hctosys --noadjfile
+ntpdate -q 0.pool.ntp.org 1.pool.ntp.org
+# Repeat these three steps a few times. If the discrepancy increases
+# consistently with ~0.5s, try to start over but pass --delay 0.0 to
+# the --systohc command and see if the situation improves.
+
+Origin: OE-lite
+Upstream-status: Inappropriate (in this form - significant changes have happened since 2.29).
```

Comment:
Well, I actually started patching upstream, which is how I found out that that patch wouldn't apply to 2.29, so I redid it against 2.29. The biggest problem is probably documentation, option naming and explaining what this is about (which I mostly left out of this one because the commit messages explain it). Also, it would actually be nice if hwclock itself figured out what an appropriate --delay value is, but that's quite hard... I'll try to send something upstream and see what they say.